api/v1alpha3/types.go

```diff
@@ -251,6 +251,29 @@ type AzureMarketplaceImage struct {
 	// time even if a new version becomes available.
 	// +kubebuilder:validation:MinLength=1
 	Version string `json:"version"`
+	// Plan specifies information about the marketplace image used to create the virtual machine.
+	// +optional
+	Plan *ImagePlan `json:"plan,omitempty"`
+}
+
+// ImagePlan specifies information about the marketplace image used to create the virtual machine. This element is
+// only used for marketplace images. Before you can use a marketplace image from an API, you must enable the
+// image for programmatic use.  In the Azure portal, find the marketplace image that you want to use and then
+// click **Want to deploy programmatically, Get Started ->**. Enter any required information and then click
+// **Save**.
+type ImagePlan struct {
```

Comment:
I wonder if this is a good user experience? Do we expect users to know how to fill this if they aren't the ones publishing the image? Or would this feature not be needed to be touched by end users in most cases?

Reply:
Another way to look at this is: would a user ever need to customize these values? If not, an alternative would be to add an optional `isThirdParty` bool in the `AzureMarketplaceImage` type which defaults to false and auto-generate the plan info if it is set to true since the values should be deterministic based on the image info.

Reply:
> Another way to look at this is: would a user ever need to customize these values? If not, an alternative would be to add an optional `isThirdParty` bool in the `AzureMarketplaceImage` type which defaults to false and auto-generate the plan info if it is set to true since the values should be deterministic based on the image info.

That seems to be inline with my comment in our conversation. It also has the least risk for both modes (upstream marketplace image versus 3rd party marketplace image).

Reply:
how would we generate the Plan fields?

Reply:
you can reuse something like this https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/320/files#diff-b2813f64759b38fd374267409eaaca65L242 

Product = offer
Name = sku
Publisher = name

I don't think we need promotion code?

Reply:
ok sounds good, will change that, didn't know we had the info to fill in the Plan